src/test/scala/com/github/symcal/ExprSpec.scala

```diff
@@ -283,4 +292,44 @@ class ExprSpec extends FlatSpec with Matchers {
     Product('x, 1, 'x, 2).subs('x, 'z) shouldEqual Product('z, 1, 'z, 2)
   }
 
+  behavior of "expand"
+
+  it should "expand monomials without change" in {
+    Const(1).expand shouldEqual Const(1)
+    ('x * 'y).expand.print shouldEqual "x * y"
+    (-'x).expand shouldEqual Minus(Var('x))
+    (-'x * 'y).expand shouldEqual Product(Minus(Var('x)), Var('y))
+  }
+
+  it should "expand sums" in {
+    ('a + 'b).expand shouldEqual Sum(Var('a), Var('b))
+    ('a - ('a - 'b)).expand shouldEqual Sum(Var('a), -Var('a), Var('b))
+    ('a - ('a - 'b)).expand.print shouldEqual "a - a + b"
+    ('a * 'a + 'a * 'b + 'b * 'a + 'b * 'b).expand.print shouldEqual "a * a + a * b + b * a + b * b"
+    (-('a + 'b)).expand shouldEqual Sum(-'a, -'b)
+    (-('a - 'b)).expand shouldEqual Sum(-'a, 'b)
+  }
+
+  it should "expand products" in {
+    (('a + 'b) * ('a + 'b)).expand.print shouldEqual "a * a + a * b + b * a + b * b"
+    (('a + 1) * ('a + 1) * ('a + 1)).expand.print shouldEqual "a * a * a + a * a + a * a + a + a * a + a + a + 1"
+
+    ('a * 'a * 'a).expand shouldEqual Product(Var('a), Var('a), Var('a))
+  }
+
+  behavior of "expand for power"
+
+  it should "compute correct multinomial coefficients" in {
+    Expr.getTermCoeffs(2, 2) shouldEqual Seq((1, Seq(2, 0)), (2, Seq(1, 1)), (1, Seq(0, 2)))
+    Expr.getTermCoeffs(2, 3) shouldEqual Seq((1, Seq(3, 0)), (3, Seq(2, 1)), (3, Seq(1, 2)), (1, Seq(0, 3)))
+  }
+
+  it should "expand powers" in {
+    (('x + 'y) #^ 0).expand.print shouldEqual "1"
+    (('x + 'y) #^ 1).expand.print shouldEqual "x + y"
+    (('x + 'y) #^ 2).expand.print shouldEqual "x^2 + x * y * 2 + y^2"
+    (('x + 'y) #^ 3).expand.print shouldEqual "x^3 + x^2 * y * 3 + x * y^2 * 3 + y^3"
```

Comment:
can we change it to have coefficients on the left of the terms, instead of on the right?